DEVELOPMENT.md

```diff
@@ -0,0 +1,32 @@
+
+# Release
+
+## How to release to PyPI
+
+#### Create PR with the new agent version
+
+- Create a pull request that updates the agent's version number in the source code [here](https://github.com/aws/amazon-codeguru-profiler-python-agent/blob/main/codeguru_profiler_agent/agent_metadata/agent_metadata.py#L12). You can use directly the editor to edit the file and create the PR.
+
+- Get approval from the [amazon-codeguru-profiler team](https://github.com/orgs/aws/teams/amazon-codeguru-profiler).
+
+- Wait until the [``Run tests`` workflow](https://github.com/aws/amazon-codeguru-profiler-python-agent/actions) passes successfully.
+
+#### Publish package to Test PyPI
+
+- Create a [new pre-release](https://github.com/aws/amazon-codeguru-profiler-python-agent/releases/new) with the agent version and details about what is changed, in the format of a changelog as [in previous releases](https://github.com/aws/amazon-codeguru-profiler-python-agent/releases). Make sure you check the box with "pre-release".
```

Comment:
Is there any check if someone didnt check the box with pre-release before getting to release?

Also are all these going to be handled by members in amazon-codeguru-profiler team only?

Reply:
1. No, and I didn't find how to do this from a quick find, it doesn't mean it is not possible though. Sounds like a nice feature. I wouldn't block on this now because even the release to test PyPI is optional, we did it only in the beginning for testing. I don't necessary see it required for a release, but it is nice to have until we make sure everything works as we expect it.

2. Yes, the repository is managed by the [amazon-codeguru-profiler team](https://github.com/orgs/aws/teams/amazon-codeguru-profiler/members), the access is setup [here](https://github.com/aws/amazon-codeguru-profiler-python-agent/settings/access).

Reply:
Cool, if the release procedure only be carried out by our team, then I am not too concern about the check 👍